### PR TITLE
[BEAM-5953] Fix DataflowRunner in Python 3 - type errors

### PR DESCRIPTION
**Pull request by markflyhigh**
Fix few places that having str <> bytes type errors related to GCP and DataflowRunner.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by markflyhigh** [comment]
+R: @charlesccychen @tvalentyn 
+cc: @aaltay

File: sdks/python/apache_beam/internal/gcp/json_value.py [hunk 1]
```diff
@@ -105,7 +105,9 @@ def to_json_value(obj, with_type=False):
     return extra_types.JsonValue(object_value=json_object)
   elif with_type:
     return to_json_value(get_typed_value_descriptor(obj), with_type=False)
-  elif isinstance(obj, (str, unicode)):
+  elif isinstance(obj, (str, unicode, bytes)):
```
Comment (tvalentyn) [inline sdks/python/apache_beam/internal/gcp/json_value.py]: I'd suggest to make a separate branch for `bytes`:
```
elif isinstance(obj, bytes):
  return extra_types.JsonValue(string_value=obj.decode('utf-8'))
```
To keep the existing behavior on Python 2, we can place this branch after 
`  elif isinstance(obj, (str, unicode)):`   branch.
```

Comment (markflyhigh) [reply sdks/python/apache_beam/internal/gcp/json_value.py]: sg. done

**Comment by aaltay** [comment]
Please ping me after tests pass for merging.

**Comment by markflyhigh** [comment]
@aaltay  Tests passed and please merge. Thanks!

**Pull request merged**